nanome/_internal/logs.py

```diff
@@ -43,57 +44,31 @@ def emit(self, record):
             pass
 
 
-class NTSFormatter(logging.Formatter):
-    """Send NTS json data with specified log level numbers."""
+class NanomeGelfHandler(graypy.handler.BaseGELFHandler):
+    """Create a GELF dictionary from a log record, including nanome-specific fields."""
 
-    datefmt = "%Y-%m-%dT%H:%M:%S"
+    def make_gelf_dict(self, record):
+        gelf_dict = self._make_gelf_dict(record)
+        gelf_dict.update({
+            'plugin_name': self._plugin.name,
+            'plugin_class': self._plugin.plugin_class.__name__,
+            'plugin_id': self._plugin._plugin_id
+        })
+        return gelf_dict
 
-    fmt = {
-        'timestamp': '%(asctime)s',
-        'msg': '%(message)s',
-        'sev': '%(levelname)s',  # Will be manually updated to val from LogType enum.
-    }
 
-    def __init__(self, fmt=None, **kwargs):
-        # Use format saved by class, so no need to pass fmt kwarg
-        fmt = json.dumps(self.fmt)
-        super(NTSFormatter, self).__init__(fmt=fmt, datefmt=self.datefmt, **kwargs)
-
-    def format(self, record):
-        msg = super(NTSFormatter, self).format(record)
-        try:
-            json_msg = json.loads(msg.replace('\n', '\\n'))
-        except json.JSONDecodeError:
-            logger.warning('JSON Decode Error in NTSFormatter')
-            updated_msg = msg
-        else:
-            # Convert timestamp to UTC
-            timestamp = json_msg['timestamp']
-            timestamp_dt = parser.parse(timestamp)
-            json_msg['timestamp'] = timestamp_dt.strftime(self.datefmt)
-
-            # Replace `sev` value with corresponding LogType from enum.
-            level_name = json_msg['sev']
-            enum_val = getattr(LogTypes, level_name)
-            json_msg['sev'] = enum_val
-            updated_msg = json.dumps(json_msg)
-        return updated_msg
-
-
-class NTSLoggingHandler(logging.Handler):
+class NTSLoggingHandler(NanomeGelfHandler):
```

Comment:
nbd, but any reason for `NTSLoggingHandler` and `NanomeGelfHandler` being separate classes? why not `NTSLoggingHandler(graypy.handler.BaseGELFHandler)`?

Reply:
I had that initially, but if we want to have the option of logging directly to Graylog w/o NTS, we'd still be able to generate the full logging dict

But after taking another look, that won't work, because we still need the plugin passed in as an arg, so I'll make them one again.